### DISCUSSION TITLE
Z position during printing is jumping around.

### DISCUSSION THREAD

**Discussion by broderp1** [mriscoc/Ender3V2S1#766] [General]
The Z position indicator in the bottom right-hand corner is jumping around.   Prior versions of firmware (Jyer's) the Z height simply showed the Z height of the current layer (0.20, 0.28 etc.) and did not bounce around.  It simply went up incrementally based on the layer height of the sliced print. (0.20, 0.40, 0.60 etc.)  

The Z icon right above is also flashing, this is also new.
Is this a bug? Can it be changed? Any ideas what this might be? 

Thank you.

**Comment by lagnat** [comment]
I had meant to ask the same question.  It is my guess that the value shown is the z height after adjusting for UBL.  It would be far more useful to show the virtual height vs the UBL adjusted height.  I also had assumed the flashing Z indicated that UBL was active.

**Reply by broderp1** [reply]
I thought this as well but for simplification left that out.   I also agree with you that it would be more useful to show the virtual height rather than the actual UBL indicator since it's all over the place and there seems to be some delay in general - just like the X and Y axis indicators.  

I would like the idea the flashing Z as an indicator that the print is using UBL.  Great idea.   I can live with that! 

Perhaps the dev or someone can assist- is there a setting to toggle this on and off or maybe this can be made into a setting or removed in a future update?

**Comment by mriscoc** [comment]
Show the axis real position is a feature implemented by user request since the 20221002 AUSTRAL SPRING version, it can be disabled by custom compiling with SHOW_REAL_POS disabled in the _configuration.h_ file.

**Comment by lagnat** [comment]
What do you think about a C-code to toggle this behavior?
